exlicit_solver_withspeedups/compute_time_deriv.jl

```diff
@@ -0,0 +1,155 @@
+    rhs::SWM_pde, 
+    params::Params, 
+    interp::Interps, 
+    grad::Derivatives, 
+    advec::Advection
+)
+
+# unpack stuff
+u = rhs.u0                          # calculate based on prognostics at
+v = rhs.v0                          # t + dt of the non-dissipative RHS
+
+(;ITu, ITv) = interp                # interpolation operators   
+(;LLu, LLv) = grad                  # gradient operators
+(;h_u, h_v) = rhs                   # diagnostic variables
+(;kinetic, kinetic_sq, Mu, Mv, nu_u, nu_v) = rhs
+(;bfric_u, bfric_v) = rhs
+(;ITu_ksq,ITv_ksq) = rhs
+(;u_t, v_t) = rhs                   # tendencies
+(;nu, bottom_drag) = params
+
+# bottom friction
+kinetic_sq .= sqrt.(kinetic)
+@inplacemul ITu_ksq = ITu * kinetic_sq
+@inplacemul ITv_ksq = ITv * kinetic_sq
+bfric_u .= bottom_drag .* ITu_ksq .* u ./ h_u
+bfric_v .= bottom_drag .* ITv_ksq .* v ./ h_v
+
+### Important: Need to apply Laplacian to nu for when nu varies spatially ##################
+# diffusion term ν∇⁴(u,v)
+@inplacemul nu_u = ITu * nu
+@inplacemul nu_v = ITv * nu
+@inplacemul Mu = LLu * u
+@inplacemul Mv = LLv * v
+Mu .*= nu_u
+Mv .*= nu_v
+
+# tendencies for bottom friction and diffusion
+u_t .= .- Mu .- bfric_u
+v_t .= .- Mv .- bfric_v 
+
+return nothing
+end
+
+function comp_u_v_eta_t(nx::Int, 
+    states_rhs::SWM_pde, 
+    params::Params, 
+    interp::Interps, 
+    grad::Derivatives, 
+    advec::Advection
+) 
+
+# unpack stuff
+u = states_rhs.u1
+v = states_rhs.v1
+eta = states_rhs.eta1
+
+(;ITu, ITv, ITq, IuT, IvT) = interp                 # interpolation operators
+(;Iuv, Ivu, Iqu, Iqv) = interp       
+(;GTx, GTy, Gux, Guy, Gvx, Gvy) = grad              # gradient operators
+(;h, h_u, h_v, h_q, U, V, p, q) = states_rhs               # diagnostic variables
+(;kinetic) = states_rhs
+(;GTx_p, GTy_p, Gux_U, Gvy_V, Gvx_v1, Guy_u1) = states_rhs
+(;adv_u, adv_v) = states_rhs
+(;IuT_u1, IvT_v1, ITu_ksq,ITv_ksq) = states_rhs
+(;u_t, v_t, eta_t) = states_rhs                            # tendencies
+(;H, coriolis, g, wind_stress) = params
+
+h .= eta .+ H 
+
+@inplacemul h_q = ITq * h
+@inplacemul h_u = ITu * h
+@inplacemul h_v = ITv * h
+U .= u .* h_u                  # volume fluxes U,V
+V .= v .* h_v 
+
+# kinetic energy u² + v²
+u2_T, u2 = IuT_u1, ITu_ksq      # reuse and rename arrays for u²
+v2_T, v2 = IvT_v1, ITv_ksq      # and v², _T is on T-grid
+u2 .= u.^2
+v2 .= v.^2
+@inplacemul u2_T = IuT * u2
+@inplacemul v2_T = IvT * v2
+kinetic .= u2_T .+ v2_T
+
+# Kloewer defined new terms q and p corresponding to potential vorticity and 
+# Bernoulli potential respectively. To avoid errors in my mimic I'm following 
+# along and doing the same 
+@inplacemul Guy_u1 = Guy * u
+@inplacemul Gvx_v1 = Gvx * v
+
+q .= (coriolis .+ Gvx_v1 .- Guy_u1) ./ h_q 
+p .= 0.5 .* kinetic .+ g .* h
+
+# For computing AL advection
+# comp_advection(nx, rhs, advec)    
+
+# Sadourny, 1975 enstrophy conserving advection scheme
+V_u = ITu_ksq                # reuse and rename array
```

Comment:
example for reusing (and renaming to avoid confusion) arrays that aren't used anymore to the end of the function